File: app/src/main/java/com/applify/securepass/VaultAdapter.java
package com.applify.securepass;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.applify.securepass.data.VaultItem;
import java.util.List;

public class VaultAdapter extends RecyclerView.Adapter<VaultAdapter.ViewHolder> {

    public interface OnItemClickListener { void onItemClick(VaultItem item); }
    public interface OnItemDeleteListener { void onDelete(VaultItem item); }

    private List<VaultItem> items;
    private OnItemClickListener listener;
    private OnItemDeleteListener deleteListener;

    public VaultAdapter(List<VaultItem> items, OnItemClickListener listener,
                        OnItemDeleteListener deleteListener) {
        this.items = items;
        this.listener = listener;
        this.deleteListener = deleteListener;
    }

    @NonNull @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(android.R.layout.simple_list_item_2, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        VaultItem item = items.get(position);
        holder.text1.setText(item.website);
        holder.text2.setText(item.username);

        // Tap: copy password to clipboard (auto‑clears)
        holder.itemView.setOnClickListener(v -> {
            ClipboardUtil.copyAndClear(v.getContext(), item.website, item.password, 30);
        });

        // Long‑press: delete entry (confirmation dialog)
        holder.itemView.setOnLongClickListener(v -> {
            new android.app.AlertDialog.Builder(v.getContext())
                    .setTitle("Delete Entry")
                    .setMessage("Delete " + item.website + "?")
                    .setPositiveButton("Delete", (dialog, which) -> {
                        if (deleteListener != null) deleteListener.onDelete(item);
                    })
                    .setNegativeButton("Cancel", null)
                    .show();
            return true;
        });
    }

    @Override public int getItemCount() { return items.size(); }

    static class ViewHolder extends RecyclerView.ViewHolder {
        TextView text1, text2;
        ViewHolder(View itemView) {
            super(itemView);
            text1 = itemView.findViewById(android.R.id.text1);
            text2 = itemView.findViewById(android.R.id.text2);
        }
    }
}

File: app/src/main/java/com/applify/securepass/VaultAdapter.kt
package com.applify.securepass

import android.R
import android.app.AlertDialog
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.applify.securepass.data.VaultItem

class VaultAdapter(
    private val items: List<VaultItem>,
    private val listener: OnItemClickListener?,
    private val deleteListener: OnItemDeleteListener?,
    private val favoriteToggleListener: OnFavoriteToggleListener?
) : RecyclerView.Adapter<VaultAdapter.ViewHolder>() {

    fun interface OnItemClickListener {
        fun onItemClick(item: VaultItem)
    }

    fun interface OnItemDeleteListener {
        fun onDelete(item: VaultItem)
    }

    fun interface OnFavoriteToggleListener {
        fun onToggle(item: VaultItem)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(android.R.layout.simple_list_item_2, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val item = items[position]
        holder.text1.text = (if (item.isFavorite) "★ " else "☆ ") + item.website
        holder.text2.text = item.username

        // Item click: show action dialog
        holder.itemView.setOnClickListener { v ->
            val options = arrayOf<CharSequence>(
                "Copy Password",
                if (item.isFavorite) "Remove from Favorites" else "Mark as Favorite",
                "Edit",
                "Delete"
            )
            AlertDialog.Builder(v.context)
                .setTitle(item.website)
                .setItems(options) { _, which ->
                    when (which) {
                        0 -> ClipboardUtil.copyAndClear(v.context, item.website, item.password, 30)
                        1 -> favoriteToggleListener?.onToggle(item)
                        2 -> listener?.onItemClick(item)
                        3 -> deleteListener?.onDelete(item)
                    }
                }
                .show()
        }

        // Remove long-press listener - now delete is in the dialog
        holder.itemView.setOnLongClickListener(null)
    }

    override fun getItemCount(): Int = items.size

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val text1: TextView = itemView.findViewById(R.id.text1)
        val text2: TextView = itemView.findViewById(R.id.text2)
    }
}